gslb/gslbutils/control_config.go

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020-2021 VMware, Inc.
+ * All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*   http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package gslbutils
+
+import (
+	gslbcs "github.com/vmware/global-load-balancing-services-for-kubernetes/internal/client/v1alpha1/clientset/versioned"
+	gdpcs "github.com/vmware/global-load-balancing-services-for-kubernetes/internal/client/v1alpha2/clientset/versioned"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type amkoControlConfig struct {
+	clientset     *kubernetes.Clientset
+	gslbClientset *gslbcs.Clientset
+	gdpClientset  *gdpcs.Clientset
+
+	amkoPodObjectMeta *metav1.ObjectMeta
+
+	amkoEventRecorder *EventRecorder
+
+	publishGDPStatus  bool
+	publishGSLBStatus bool
+}
+
+var amkoControlConfigInstance *amkoControlConfig
+
+func AMKOControlConfig() *amkoControlConfig {
+	if amkoControlConfigInstance == nil {
```

Comment:
This is not a thread-safe way of initializing a singleton variable as this operation and the below operation is not atomic. Please use the sync package's once.Do() way of initializing singletons (https://pkg.go.dev/sync#Once.Do).